src/System.Text.Json/docs/ReferenceHandling_spec.md

```diff
@@ -0,0 +1,433 @@
+}
+```
+
+Output:
+```jsonc
+{
+    "$id": "1",
+    "Name": "Angela",
+    "Manager": {
+        "$id": "2",
+        "Name": "Bob",
+        "Subordinates": { //Note how the Subordinates' square braces are replaced with curly braces in order to include $id and $values properties, $values will now hold whatever value was meant for the Subordinates list.
+            "$id": "3",
+            "$values": [
+                {  //Note how this object denotes reference to Angela that was previously serialized.
+                    "$ref": "1"
+                }
+            ]
+        }            
+    }
+}
+```
+
+## Using Preserve on Deserialize
+On System.Text.Json:
+```cs
+public static void ReadJsonWithPreservedReferences(){
+    string json = 
+    @"{
+        ""$id"": ""1"",
+        ""Name"": ""Angela"",
+        ""Manager"": {
+            ""$id"": ""2"",
+            ""Name"": ""Bob"",
+            ""Subordinates"": {
+                ""$id"": ""3"",
+                ""$values"": [
+                    { 
+                        ""$ref"": ""1"" 
+                    }
+                ]
+            }            
+        }
+    }";
+
+    var options = new JsonSerializerOptions
+    {
+        ReferenceHandling = ReferenceHandling.Preserve
+    };
+
+    Employee angela = JsonSerializer.Deserialize<Employee>(json, options);
+    Console.WriteLine(object.ReferenceEquals(angela, angela.Manager.Subordinates[0])); //prints: true.
+}
+```
+
+On Newtonsoft's Json.Net:
+```cs
+public static void ReadJsonWithPreservedReferences(){
+    string json = 
+    @"{
+        ""$id"": ""1"",
+        ""Name"": ""Angela"",
+        ""Manager"": {
+            ""$id"": ""2"",
+            ""Name"": ""Bob"",
+            ""Subordinates"": {
+                ""$id"": ""3"",
+                ""$values"": [
+                    { 
+                        ""$ref"": ""1"" 
+                    }
+                ]
+            }            
+        }
+    }";
+
+    var options = new JsonSerializerSettings
+    {
+        MetadataPropertyHanding = MetadataPropertyHandling.Default //Json.Net reads metadata by default, just setting the option for ilustrative purposes.
+    };
+
+    Employee angela = JsonConvert.DeserializeObject<Employee>(json, settings);
+    Console.WriteLine(object.ReferenceEquals(angela, angela.Manager.Subordinates[0])); //prints: true.
+}
+```
+
+# Ground rules
+## Unsupported types
+Basically, any type that uses a `EnumerableConverter` and tries to be preserved will throw.
+
+* **Immutable types**: i.e: `ImmutableList` and `ImmutableDictionary`
+* **System.Array**
+
+Aside from above, the Deserializer will throw when a CLR value type is passed as preserved when `ReferenceHandling.Preserve` is set. 
+
+As a rule of thumb, we should throw on all cases where the payload contains metadata that is impossible to create with the `JsonSerializer`, however, this conflicts with feature parity in Json.Net; those scenarios are described below.
```

Comment:
```suggestion
As a rule of thumb, we throw on all cases where the JSON payload being read contains any metadata that is impossible to create with the `JsonSerializer` (i.e. it was hand modified). However, this conflicts with feature parity in Newtonsoft.Json; those scenarios are described below.
```

Reply:
This doesn't belong under the `Unsupported types` imo, but rather should go under the `Ground rules` heading, on its own.